src/lib/Runner.ts

```diff
@@ -223,32 +223,34 @@ export class Runner {
         this.config.mergeSettings &&
         this.config.mergeSettings.skipBuildOnDependentsAwaitingMerge;
       const pullRequestId = landRequest.pullRequestId;
-      Logger.verbose('Attempting merge pull request', {
+      Logger.info('Triggering merge attempt', {
         namespace: 'lib:runner:moveFromAwaitingMerge',
         pullRequestId,
         landRequestId: landRequest.id,
         lockId,
       });
-      await this.client.mergePullRequest(landRequestStatus, { skipCI });
-      Logger.info('Successfully merged PR', {
-        namespace: 'lib:runner:moveFromAwaitingMerge',
-        landRequestId: landRequest.id,
-        pullRequestId,
-        lockId,
-      });
 
-      const end = Date.now();
-      const queuedDate = await this.getLandRequestQueuedDate(landRequest.id);
-      const start = queuedDate!.getTime();
-      eventEmitter.emit('PULL_REQUEST.MERGE.SUCCESS', {
-        landRequestId: landRequestStatus.requestId,
-        pullRequestId: landRequest.pullRequestId,
-        commit: landRequest.forCommit,
-        sourceBranch: pullRequest.sourceBranch,
-        targetBranch: pullRequest.targetBranch,
-        duration: end - start,
-      });
-      return landRequest.setStatus('success');
+      await landRequest.setStatus('merging');
+      this.client.mergePullRequest(
+        landRequestStatus,
+        async () => {
+          const end = Date.now();
+          const queuedDate = await this.getLandRequestQueuedDate(landRequest.id);
+          const start = queuedDate!.getTime();
+          eventEmitter.emit('PULL_REQUEST.MERGE.SUCCESS', {
+            landRequestId: landRequestStatus.requestId,
+            pullRequestId: landRequest.pullRequestId,
+            commit: landRequest.forCommit,
+            sourceBranch: pullRequest.sourceBranch,
+            targetBranch: pullRequest.targetBranch,
+            duration: end - start,
+          });
+          landRequest.setStatus('success');
+        },
+        { skipCI },
+      );
+
+      return true;
     } catch (err) {
       eventEmitter.emit('PULL_REQUEST.MERGE.FAIL', {
```

Comment:
Will this `catch` block catch errors thrown in the `mergePullRequest` function and/or the callback? We might have to add a `.catch()` directly to the function call.